element_array_ephys/ephys_no_curation.py

```diff
@@ -0,0 +1,863 @@
+    -> EphysRecording
+    ---
+    lfp_sampling_rate: float   # (Hz)
+    lfp_time_stamps: longblob  # (s) timestamps with respect to the start of the recording (recording_timestamp)
+    lfp_mean: longblob         # (uV) mean of LFP across electrodes - shape (time,)
+    """
+
+    class Electrode(dj.Part):
+        definition = """
+        -> master
+        -> probe.ElectrodeConfig.Electrode  
+        ---
+        lfp: longblob               # (uV) recorded lfp at this electrode 
+        """
+
+    # Only store LFP for every 9th channel, due to high channel density,
+    # close-by channels exhibit highly similar LFP
+    _skip_channel_counts = 9
+
+    def make(self, key):
+        acq_software, probe_sn = (EphysRecording
+                                  * ProbeInsertion & key).fetch1('acq_software', 'probe')
+
+        electrode_keys, lfp = [], []
+
+        if acq_software == 'SpikeGLX':
+            spikeglx_meta_filepath = get_spikeglx_meta_filepath(key)
+            spikeglx_recording = spikeglx.SpikeGLX(spikeglx_meta_filepath.parent)
+
+            lfp_channel_ind = spikeglx_recording.lfmeta.recording_channels[
+                          -1::-self._skip_channel_counts]
+
+            # Extract LFP data at specified channels and convert to uV
+            lfp = spikeglx_recording.lf_timeseries[:, lfp_channel_ind]  # (sample x channel)
+            lfp = (lfp * spikeglx_recording.get_channel_bit_volts('lf')[lfp_channel_ind]).T  # (channel x sample)
+
+            self.insert1(dict(key,
+                              lfp_sampling_rate=spikeglx_recording.lfmeta.meta['imSampRate'],
+                              lfp_time_stamps=(np.arange(lfp.shape[1])
+                                               / spikeglx_recording.lfmeta.meta['imSampRate']),
+                              lfp_mean=lfp.mean(axis=0)))
+
+            electrode_query = (probe.ProbeType.Electrode
+                               * probe.ElectrodeConfig.Electrode
+                               * EphysRecording & key)
+            probe_electrodes = {
+                (shank, shank_col, shank_row): key
+                for key, shank, shank_col, shank_row in zip(*electrode_query.fetch(
+                    'KEY', 'shank', 'shank_col', 'shank_row'))}
+
+            for recorded_site in lfp_channel_ind:
+                shank, shank_col, shank_row, _ = spikeglx_recording.apmeta.shankmap['data'][recorded_site]
+                electrode_keys.append(probe_electrodes[(shank, shank_col, shank_row)])
+        elif acq_software == 'Open Ephys':
+            sess_dir = find_full_path(get_ephys_root_data_dir(),
+                                      get_session_directory(key))
+            loaded_oe = openephys.OpenEphys(sess_dir)
+            oe_probe = loaded_oe.probes[probe_sn]
+
+            lfp_channel_ind = np.arange(
+                len(oe_probe.lfp_meta['channels_ids']))[-1::-self._skip_channel_counts]
```

Comment:
I think this code is equivalent to
```suggestion
            lfp_channel_ind = np.r_[len(oe_probe.lfp_meta['channels_ids']),0:-self._skip_channel_counts]
```